doc/remailers.rst

```diff
@@ -0,0 +1,129 @@
+Autocrypt and Mailing Lists
+====
+
+General Considerations
+----
+
+"Mailing lists" are a special use case of e-mail, where some address is managed
+not as an inbox that stores e-mail for retrieval, but instead forwards all
+messages to a configurable list of subscribers. For the case of end-to-end
+encryption, the introduction of a relaying agent brings up the question of what
+constitutes an "Endpoint".
+
+For users with a typical intuition of end-to-end encryption, the expected end
+points of end-to-end encrypted mailing lists will likely be the individual
+subscribers. This is very difficult to achieve on top of existing e-mail
+infrastructure, since it requires either negotiation between all endpoints, or
+use of experimental encryption techniques such as homomorphic or attribute-based
+encryption.
+
+A compromise solution is to encrypt from the sender only to the remailer, which
+then decrypts the message and individually encrypts to all subscribers. This
+method violates strict end-to-end confidentiality, since the remailer can access
+to the forwarded plaintext. However, the remailer will only need to handle
+sensitive data transiently, which means messages are secure as long as the
+remailer is honest at the time of sending.
+
+This document focuses on seamless support for re-encryption based end-to-end
+security in existing remailers, in keeping with the more general Autocrypt
+approach.
+
+Autocrypt-enabled Remailer
+----
+
+An Autocrypt-enabled remailer maintains an Autocrypt key of its own. During
+regular operation, it collects Autocrypt-headers of all list participants via
+incoming mail, to keep track of their keys. Then, depending on the availability
+of encryption keys for participants, the remailer injects its own
+Autocrypt-headers into outgoing mail to signal availability of encryption.
+
+More specifically, an Autocrypt-enabled mailing list adds three behaviors to its
+remailing logic:
+
+1) For all incoming messages, keep track of Autocrypt-headers. This mechanism is
+   the same that is used by clients, and requires no modifications.
+2) Inject an Autocrypt-header into outgoing messages. This header serves the
+   same purpose as usual, but adds some details for the mailing list use case.
+   The header might only be injected under certain conditions, e.g. if
+   encryption keys are available for a certain number of subscribers.
+3) When receiving an encrypted message, decrypt it and re-encrypt for individual
+   delivery to each Autocrypt-enabled subscriber. Subscribers for whom no keys
+   are available do not receive the message, but optionally a notification of
+   this circumstance (see below).
```

Comment:
This constitutes for me a particular policy of running a mailing list, certainly an interesting one.  It means that users who are autocrypt enabled can send each other confidential information, e.g. roomcodes for an event, passwords for sites , etc.